evap/evaluation/models.py

```diff
@@ -1,34 +1,332 @@
+
+        def choice_to_display(field, choice):  # does not support nested choices
+            return next(filter(lambda t: t[0] == choice, field.choices), (choice, choice))[1]
+
+        for field_name, actions in data.items():
+            field = model._meta.get_field(field_name)
+            try:
+                label = getattr(field, "verbose_name", field_name).capitalize()
+            except FieldDoesNotExist:
+                label = field_name.capitalize()
+
+            if field.many_to_many or field.many_to_one or field.one_to_one:
+                # Convert item values from primary keys to string-representation for special fields
+                related_ids = itertools.chain(*actions.values())
+                related_objects = field.related_model.objects.filter(pk__in=related_ids)
+                bool(related_objects)  # force queryset evaluation
+                for field_action_type, primary_keys in actions.items():
+                    items = [self._pk_to_string_representation(key, field, related_objects) for key in primary_keys]
+                    fields[field_name].append(FieldAction(label, field_action_type, items))
+            elif hasattr(field, "choices") and field.choices:
+                for field_action_type, items in actions.items():
+                    fields[field_name].append(FieldAction(
+                        label, field_action_type, [choice_to_display(field, item)for item in items])
+                    )
+            else:
+                for field_action_type, items in actions.items():
+                    fields[field_name].append(FieldAction(label, field_action_type, items))
+        return dict(fields)
+
+    def display(self):
+        if self.action_type not in ("change", "create", "delete"):
+            raise ValueError("Unknown action type: '{}'!".format(self.action_type))
+
+        field_data = json.loads(self.data)
+
+        if self.action_type == 'change':
+            if self.content_object:
+                message = _("The {cls} {obj} was changed.")
+            else:
+                message = _("A {cls} was changed.")
+        elif self.action_type == 'create':
+            if self.content_object:
+                message = _("The {cls} {obj} was created.")
+            else:
+                message = _("A {cls} was created.")
+        elif self.action_type == 'delete':
+            message = _("A {cls} was deleted.")
+
+        message = message.format(
+            cls=self.content_type.model_class()._meta.verbose_name_raw,
+            obj=f"\"{self.content_object!s}\"" if self.content_object else "",
+        )
+
+        return render_to_string("log/changed_fields_entry.html", {
+            'message': message,
+            'fields': self._evaluation_log_template_context(field_data),
+        })
+
+
+def log_serialize(obj):
+    if obj is None:
+        return ""
+    if isinstance(obj, (date, time, datetime)):
+        return localize(obj)
```

Comment:
JSON doesn't have a datetime datatype, so it would store a datetime as an ISO-string. Now instead of building special serialization logic for that, I just store it as a appropriately formatted string here already. Afaik, we use the same formats for german and english, so this should be fine.